### ISSUE TITLE
[V3 Streams] Youtube alerts added by ID have NoneType name

### ISSUE THREAD

**Issue by JJW8071** [Cog-Creators/Red-DiscordBot#1999]
# Command bugs

Red v3.0.0.b18

#### Command name

[p]streamalert youtube UC79_r-MK8-bmJCYmyzAkvPw

#### What cog is this command from?

Streams 

#### What were you expecting to happen?

Add non-partner youtube channel ("Cade") to streamalert settings.
Add <id: channel-name>"Cade"</channel-name> name to streams settings in lieu of a Partner-Name.

#### What actually happened?
**Streams cog is completely disabled**

If the Youtube Channel ID is used <UC....xzy> instead of a Partner channel name AND if the youtube channel is *not* a Youtube Partner, then Streams sets the channel name as 'null'.

```
Exception during loading of cog
Traceback (most recent call last):
  File "C:\Users\jason\AppData\Local\Programs\Python\Python36\lib\site-packages\redbot\core\core_commands.py", line 87, in _load
    await bot.load_extension(spec)
  File "C:\Users\jason\AppData\Local\Programs\Python\Python36\lib\site-packages\redbot\core\bot.py", line 222, in load_extension
    await lib.setup(self)
  File "C:\Users\jason\AppData\Local\Programs\Python\Python36\lib\site-packages\redbot\cogs\streams\__init__.py", line 6, in setup
    await cog.initialize()
  File "C:\Users\jason\AppData\Local\Programs\Python\Python36\lib\site-packages\redbot\cogs\streams\streams.py", line 71, in initialize
    self.streams = await self.load_streams()
  File "C:\Users\jason\AppData\Local\Programs\Python\Python36\lib\site-packages\redbot\cogs\streams\streams.py", line 618, in load_streams
    return [x for x in streams if not (x.name.lower() in seen or seen_add(x.name.lower()))]
  File "C:\Users\jason\AppData\Local\Programs\Python\Python36\lib\site-packages\redbot\cogs\streams\streams.py", line 618, in <listcomp>
    return [x for x in streams if not (x.name.lower() in seen or seen_add(x.name.lower()))]
AttributeError: 'NoneType' object has no attribute 'lower'
```

#### How can we reproduce this issue?

[p]streamalert youtube UC79_r-MK8-bmJCYmyzAkvPw
[p]reload streams
[error]

**Comment by mikeshardmind** [comment]
Can repro on both b18 and V3/dev This makes the cog completely inoperable.

This should be a reasonably easy fix on preventing the issue on load by making this NoneType safe.

More work will need to be done to figure out how/why we are storing a NoneType here to begin with though.

**Comment by Tobotimus** [comment]
Surely the traceback is different on V3/develop though? Since #1994 has been merged

**Comment by Tobotimus** [comment]
If this is an easy fix we'll try to fit it in beta 19, I'm willing to accept a PR for it after our cut-off at 23:59 EST Friday

**Comment by mikeshardmind** [comment]
@Tobotimus Not related to the token. This has to do with a failure to get stream info in another case. The "easy" fix is a NoneType filter, but that doesn't fix the root problem, and will cause stream alerts for those channels to fail silently. This is still better than breaking all stream alerts to accomplish this.

**Comment by Tobotimus** [comment]
This'll no longer be fatal to the Streams cog in beta 19 after the changes in #1994. However, the name for a youtube stream added by ID is still being set to `None`. This issue will remain open until that is fixed.

**Comment by Tobotimus** [comment]
Thanks for reporting this issue @JasonJW.

Resolved by #2047.
